### PR TITLE
Update for ECE 3.X

### PR DESCRIPTION
**Pull request by ianrscottexp**
Added Features
- Use ECE 3.X REST API calls
- Support Deletion of Elastic Instances
- Support ML elastic resources

### CODE REVIEW

File: plugins/module_utils/ece.py [hunk 1]
```diff
@@ -110,39 +174,186 @@ def get_snapshot_repo_by_name(self, repo_name):
...
+    #endpoint = f'clusters/elasticsearch/{cluster_id}/_shutdown'
+    target_deployment_object = self.get_deployment_byid(deployment_id)
+    resources = target_deployment_object['resources'][resource_kind]
+    for resource in resources: 
+      target_resource_ref_id = resource['ref_id']
+      endpoint = f'deployments/{deployment_id}/{resource_kind}/{target_resource_ref_id}/_shutdown'
+      stop_result = self.send_api_request(endpoint, 'POST')
+      wait_result = self.wait_for_cluster_state(deployment_id, resource_kind, target_resource_ref_id,'stopped', self.completion_timeout)
+    if not wait_result:
+      self.module.fail_json(msg=f'failed to stop cluster {self.cluster_name}')
```
Comment (mgaruccio) [inline plugins/module_utils/ece.py:358]: Am I missing something or would this mean that any time you set `wait_result` to false the module would return a failure, regardless of the success or failure of the command?

**Review by mgaruccio — CHANGES_REQUESTED** [review]
I know this has been tested so in-general looks good to me.  Only have the one note around the `wait_for_completion` but if I just misunderstood there then this looks perfect.

File: plugins/module_utils/ece.py (hunk 1 above)
Comment (ianrscottexp) [reply plugins/module_utils/ece.py:358]: The wait_result value  is a result of the wait_for_cluster_state.  I assume the intention was to wait for 600 seconds by default and if it still doesn't stop to return a failure.  I didn't really create that part of the code I was just trying to get the status returned properly. If you want it to behave differently, let me know what you want the it to be. I believe this is true for creating the elastic instance as well

Comment (mgaruccio) [reply plugins/module_utils/ece.py:358]: ohh yea I was reading the code wrong, I see what that's doing now.  looks good to me

**Review by mgaruccio — APPROVED** [review]
lgtm

**Review by exp-hc — APPROVED** [review]
LGTM

**Review by jesselarrabee — APPROVED** [review]

**Pull request merged**